tests/FakeItEasy.Tests/Configuration/RuleBuilderTests.cs

```diff
@@ -32,6 +34,30 @@ public RuleBuilderTests()
             this.builder = this.CreateBuilder();
         }
 
+        public static IEnumerable<object[]> CallSpecificationActionsForVoid =>
+            TestCases.FromObject<Action<IVoidArgumentValidationConfiguration>>(
+                configuration => configuration.WhenArgumentsMatch(args => true));
+
+        public static IEnumerable<object[]> CallSpecificationActionsForNonVoid =>
+            TestCases.FromObject<Action<IAnyCallConfigurationWithReturnTypeSpecified<int>>>(
+                configuration => configuration.WhenArgumentsMatch(args => true),
+                configuration => configuration.Where(call => true));
+
+        public static IEnumerable<object[]> BehaviorDefinitionActionsForVoid =>
+            TestCases.FromObject<Action<IVoidArgumentValidationConfiguration>>(
+                configuration => configuration.CallsBaseMethod(),
+                configuration => configuration.DoesNothing(),
+                configuration => configuration.Throws<Exception>(),
+                configuration => configuration.Invokes(DoNothing),
+                configuration => configuration.AssignsOutAndRefParametersLazily(_ => new object[0]));
+
+        public static IEnumerable<object[]> BehaviorDefinitionActionsForNonVoid =>
+            TestCases.FromObject<Action<IAnyCallConfigurationWithReturnTypeSpecified<int>>>(
+                configuration => configuration.CallsBaseMethod(),
+                configuration => configuration.Throws<Exception>(),
+                configuration => configuration.Invokes(DoNothing),
+                configuration => configuration.ReturnsLazily(_ => 0));

```

Comment:
I was going to say that I can live without `AssignsOutAndRefParametersLazily`, but in case you accidentally forgot, you could add.
Then I realized that _you can't_, because unlike `Invokes`, as we were talking about earlier, you can't assign the outs and refs until you return something. Or similar.